imap/http_caldav.c

```diff
@@ -6190,7 +6190,18 @@ static int proppatch_timezone(xmlNodePtr prop, unsigned set,
                                                  httpd_userid, &pctx->buf);
             if (!r) {
                 /* Set CALDAV:calendar-timezone */
-                proppatch_todb(prop, set, pctx, propstat, (void *) tz);
+                r = proppatch_todb(prop, set, pctx, propstat, (void *) tz);
+
+                if (!r) {
+                    /* Update alarms on floating time events */
+                    icaltimezone *ical_tz = icaltimezone_new();
+
+                    icaltimezone_set_component(ical_tz, icalcomponent_clone(vtz));
+                    caldav_alarm_update_floating(pctx->mailbox,
```

Comment:
Oh yeah, I almost forgot.  You can't do it at this layer.  This needs to be triggered down in the annotations layer with a string match on the annotation, because otherwise the replica calalarm database won't be updated.

Reply:
I moved the updating into a annotation_set_timezone() callback and used the annotate_state_store() API which uses the callbacks (unlike annotate_state_writemask())